build.gradle

```diff
@@ -516,14 +516,20 @@ def numbersToLetters(String version) {
 
 def getStandalonePackageIdentifier(ccDomainSafe, isConsumerApp) {
     if (project.hasProperty('application_name')) {
-        return ccDomainSafe + "." + application_name.replaceAll(" ", "").toLowerCase();
+        return ccDomainSafe + "." + cleanseAppNameForPackageIdentifier(application_name)
     } else if ("true".equals(isConsumerApp)) {
         throw new InvalidUserDataException("An application_name property must be provided for consumer app standalone builds")
     } else {
         return ccDomainSafe;
     }
 }
 
+
+def cleanseAppNameForPackageIdentifier(appName) {
+    return appName.replaceAll("[ '.:]", "").replaceAll("\\\\", "").toLowerCase()
```

Comment:
No great reason, just because I found it confusing that I needed 4 '\' characters to actually represent a regex for '\', so I left it separate because I wasn't sure how they'd interact